edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/CloudProxyTest.cs

```diff
@@ -216,29 +243,30 @@ public async Task TestHandleNonRecoverableExceptions(Type exceptionType)
             client.VerifyAll();
         }
 
-        static async Task CheckMessageInEventHub(IList<IMessage> sentMessages, DateTime startTime)
+        static async Task CheckMessageInEventHub(Dictionary<string, IList<IMessage>> sentMessagesByDevice, DateTime startTime)
         {
             string eventHubConnectionString = await SecretsHelper.GetSecretFromConfigKey("eventHubConnStrKey");
             var eventHubReceiver = new EventHubReceiver(eventHubConnectionString);
-            var cloudMessages = new List<EventData>();
-            bool messagesFound = false;
-            // Add retry mechanism to make sure all the messages sent reached Event Hub.
-            for (int i = 0; i < EventHubMessageReceivedRetry; i++)
+            var receivedMessagesByDevice = new Dictionary<string, List<EventData>>();
+
+            await Task.Delay(TimeSpan.FromSeconds(30));
```

Comment:
since you changed to wait a fixed time once, please move this line at the top and add comment to explain why you do this.